code/turf/turf_flock.dm

```diff
@@ -356,6 +356,19 @@
 		else if(!isfeathertile(newloc))
 			F.end_floorrunning()
 
+/turf/simulated/wall/auto/feather/Bumped(AM)
+	. = ..()
+	if(istype(AM, /mob/living/critter/flock/drone))
+		var/mob/living/critter/flock/drone/F = AM
+		if(F.floorrunning || (F.can_floorrun && F.resources >= 1))
+			if(!F.client || (F.client && F.client.check_key(KEY_RUN))) //ai doesn't have to hold shift to wallrun, people do
```

Comment:
If bumped can be called in more cases than the player purposefully bumping into wall, feel like is_npc and dormant checks might be safer than a no client check?

Reply:
will fix